Sources/Misc/Obsoletions.swift

```diff
@@ -470,6 +470,16 @@ public extension Purchases {
         fatalError()
     }
 
+    @available(iOS, obsoleted: 1, message: "Use PurchasesDelegate.purchases(_:isReadyForPromotedProduct:purchase:)")
+    @available(tvOS, obsoleted: 1, message: "Use PurchasesDelegate.purchases(_:isReadyForPromotedProduct:purchase:)")
+    @available(watchOS, obsoleted: 1, message: "Use PurchasesDelegate.purchases(_:isReadyForPromotedProduct:purchase:)")
+    @available(macOS, obsoleted: 1, message: "Use PurchasesDelegate.purchases(_:isReadyForPromotedProduct:purchase:)")
+    @available(macCatalyst, obsoleted: 1, message: "Use PurchasesDelegate.purchases(_:isReadyForPromotedProduct:purchase:)")
+    @objc func shouldPurchasePromoProduct(_ product: StoreProduct,
```

Comment:
I marked this obsoleted instead of deprecated. I don't think it made sense for users to call it anyway, it was more of a programmer error, so better to fail. Breaking compilation in this case I think is warranted.

Reply:
> This was never meant to be `public`. Use `PurchasesDelegate.purchases(_:isReadyForPromotedProduct:purchase:)`
?

Reply:
LGTM

Reply:
Changed.